### DISCUSSION TITLE
Using PyBaMM to solve specific set of electrolyte equations

### DISCUSSION THREAD

**Discussion by isaacsquires** [pybamm-team/PyBaMM#2160] [Q&A] [answered]
Hi everyone,

I have the following system of equations.

<img width="430" alt="Screenshot 2022-07-06 at 18 28 00" src="https://user-images.githubusercontent.com/55202483/177749539-d93a5fdc-fbfb-4f9e-ac87-7be48f2e2b84.png">

It is a symmetrical cell system with no separator, with two electrodes with infinite lithium resource and a sort of Butler-Volmer type exchange at the interface. $\Phi$ is prescribed, the four variable I am solving for are $c_0, \phi_0, j_{eff}, F_{eff}$ and both $D_{eff}, k_{eff}$ are functions of $c_0$, everything else is constant. Also, this is a porous electrode which is captured in effective diffusivity and conductivity I want to use PyBaMM to solve these equations but I don't want the entire DFN, just the electrolyte part, and also my equations are slightly different. What is the best way to go about this?

I was thinking of starting with the electrolyte submodel and adapting the reaction term.

**Comment by brosaplanella** [comment, marked as answer]
From the looks of it, it seems you will "only" have to call `BaseBatteryModel` and define the electrolyte submodels only + a custom submodel for the interface. This [notebook on submodels](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/models/using-submodels.ipynb) should be a good starting point.
